### PR TITLE
feat: adds ability to upload files from a remote url

### PR DESCRIPTION
**Pull request by PatrikKozak**
## Description

`feat`: adds ability to upload files from a remote url

Adds new button to uploads labeled `Paste URL` 

Screenshot 2024-07-08 at 10 46 14 AM

Upon clicking it, a modal with an input field will appear to where one can input a remote url of an image.

Screenshot 2024-07-08 at 10 46 22 AM

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes

### CODE REVIEW

File: test/fields/collections/Upload/e2e.spec.ts [hunk 1]
```diff
@@ -81,11 +81,41 @@ describe('Upload', () => {
...
+
+    await page.goto(url.create)
+
+    const pasteURLButton = page.locator('.file-field__upload .dropzone__file-button', {
+      hasText: 'Paste URL',
+    })
+    await pasteURLButton.click()
+
+    const remoteImage =
+      'https://fastly.picsum.photos/id/11/2500/1667.jpg?hmac=xxjFJtAPgshYkysU_aqx2sZir-kIOjNR9vx0te7GycQ'
```
Comment (denolfe) [inline test/fields/collections/Upload/e2e.spec.ts:99]: Do we think this URL will be available long-term? If not, we might want to use something in our control like the og-image from our site.

File: packages/payload/src/uploads/types.ts [hunk 2]
```diff
@@ -82,6 +82,11 @@ export type UploadConfig = {
    * - A function that generates a fully qualified URL for the thumbnail, receives the doc as the only argument.
    **/
   adminThumbnail?: GetAdminThumbnail | string
+  /**
+   * Enables the ability to upload a file from a remote URL instead of selecting a file from your system.
+   * @default false
+   */
+  allowRemoteUpload?: boolean
```
Comment (denolfe) [inline packages/payload/src/uploads/types.ts:89]: Is there any reason this wouldn't be enabled by default (and also sanitized as true)?

**Review by denolfe — CHANGES_REQUESTED** [review]

File: packages/payload/src/uploads/types.ts (hunk 2 above)
Comment (PatrikKozak) [reply packages/payload/src/uploads/types.ts:89]: I suppose there would be no harm to default it to true

File: test/fields/collections/Upload/e2e.spec.ts (hunk 1 above)
Comment (PatrikKozak) [reply test/fields/collections/Upload/e2e.spec.ts:99]: I was thinking the same thing - I will update

**Review by denolfe — APPROVED** [review]

**Pull request merged**